src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -94,6 +103,8 @@ trait Blockchain {
 
   def save(hash: ByteString, evmCode: ByteString): Unit
 
+  def save(node: MptNode): Unit
```

Comment:
I think the state is part of blockchain because it is directly linked with blockchain, any change in blockchain will affect the state in some way, either ether transfer or contract creation/call, so I see no point in splitting those two things